x-pack/plugins/infra/public/alerting/log_threshold/components/alert_details_app_section/components/explain_log_rate_spike.tsx

```diff
@@ -94,21 +94,50 @@ export const ExplainLogRateSpikes: FC<AlertDetailsExplainLogRateSpikesSectionPro
     }
   }, [rule, alert, dataViews, logViews]);
 
+  // Identify `intervalFactor` to adjust time ranges based on alert settings.
+  // The default time ranges for `initialAnalysisStart` are suitable for a `1m` lookback.
+  // If an alert would have a `5m` lookback, this would result in a factor of `5`.
+  const lookbackDuration =
+    alert.fields['kibana.alert.rule.parameters'] &&
+    alert.fields['kibana.alert.rule.parameters'].timeSize &&
+    alert.fields['kibana.alert.rule.parameters'].timeUnit
+      ? moment.duration(
+          alert.fields['kibana.alert.rule.parameters'].timeSize as number,
+          alert.fields['kibana.alert.rule.parameters'].timeUnit as any
+        )
+      : moment.duration(1, 'm');
+  const intervalFactor = Math.max(1, lookbackDuration.asSeconds() / 60);
+
   const alertStart = moment(alert.start);
   const alertEnd = alert.fields[ALERT_END] ? moment(alert.fields[ALERT_END]) : undefined;
 
   const timeRange = {
-    min: alertStart.clone().subtract(20, 'minutes'),
-    max: alertEnd ? alertEnd.clone().add(5, 'minutes') : moment(new Date()),
+    min: alertStart.clone().subtract(20 * intervalFactor, 'minutes'),
+    max: alertEnd ? alertEnd.clone().add(5 * intervalFactor, 'minutes') : moment(new Date()),
```

Comment:
For `max`, alert end + `intervalFactor * 1` should be enough. What do you think? Should we also add check if `max` is after current time, then `max` should be current time.

If the alert is active since long time, then showing data of (alert "start" + `3 * intervalFactor`) should also be enough to analyse. And similarly deviation and baseline can be shortened. But let's evaluate that later.

Reply:
Tweaked the time ranges in [34a396f](https://github.com/elastic/kibana/pull/160150/commits/34a396f431630a82327eafc20e728db35de8740e). This is the example from the description after the alert recovered:

Before:

<img width="1428" alt="image" src="https://github.com/elastic/kibana/assets/230104/f6da5832-9a16-4467-9975-6fd567547728">

After:

<img width="1424" alt="image" src="https://github.com/elastic/kibana/assets/230104/50d09f73-36c4-4228-88f6-8e309e793af5">

Alternative with deviation until the end of the alert, when it's an alert that recovered (-1 lookback segment):

<img width="1433" alt="image" src="https://github.com/elastic/kibana/assets/230104/e9bde63b-0dbe-4af3-b97f-dc982838c9a3">

IMHO The last might make the most sense when explaining the auto-select of time ranges to users saying we're using the full alert range we get from the metadata instead of some arbritray interval we've chosen. What do you think?

Reply:
Pushed an update in [86ff889](https://github.com/elastic/kibana/pull/160150/commits/86ff889320e3936e5cf7b9420de23ab05ed05bda) to match the last screenshot. Just moved the inline code to get `deviationMax` to a helper function to avoid a double ternary.